Library/Homebrew/formula_support.rb

```diff
@@ -28,13 +28,14 @@ def valid?
 
   def to_s
     return @explanation unless @explanation.empty?
+    os_name = OS::Mac.os_name(ENV["HOMEBREW_OSX_VERSION"])
```

Comment:
@MikeMcQuaid If necessary I can also move other `OS X` referencing user-facing messages to `/extend/os/mac`.